contracts/src/DataHavenServiceManager.sol

```diff
@@ -232,6 +236,48 @@ contract DataHavenServiceManager is ServiceManagerBase, IDataHavenServiceManager
         return address(_snowbridgeGateway);
     }
 
+    // ============ Rewards Submitter Functions ============
+
+    /// @inheritdoc IDataHavenServiceManager
+    function submitRewards(
+        IRewardsCoordinatorTypes.OperatorDirectedRewardsSubmission calldata submission
+    ) external override onlyRewardsInitiator {
+        // Calculate total amount for event
+        uint256 totalAmount = 0;
+        for (uint256 i = 0; i < submission.operatorRewards.length; i++) {
+            totalAmount += submission.operatorRewards[i].amount;
+        }
+
+        // Approve RewardsCoordinator to spend tokens
+        submission.token.safeIncreaseAllowance(address(_rewardsCoordinator), totalAmount);
+
+        // Wrap in array for RewardsCoordinator
+        IRewardsCoordinatorTypes.OperatorDirectedRewardsSubmission[] memory submissions =
+            new IRewardsCoordinatorTypes.OperatorDirectedRewardsSubmission[](1);
+        submissions[0] = submission;
+
+        // Submit to EigenLayer RewardsCoordinator
+        OperatorSet memory operatorSet = OperatorSet({avs: address(this), id: VALIDATORS_SET_ID});
+        _rewardsCoordinator.createOperatorDirectedOperatorSetRewardsSubmission(
+            operatorSet, submissions
+        );
+
+        emit RewardsSubmitted(totalAmount, submission.operatorRewards.length);
+    }
+
+    /// @notice Sets the rewards initiator address (overrides deprecated base implementation)
+    /// @param newRewardsInitiator The new rewards initiator address
+    /// @dev Only callable by the owner
+    function setRewardsInitiator(
```

Comment:
If for any reason this rewardsInitiator is misconfigured (or for example the zero address), every rewards submission tx from the bridge reverts with `OnlyRewardsInitiator()`. In this scenario `on_era_end` will still mint inflation and we will still mint gas from the failed `submitRewards()`. Maybe we should add an extra check here or in `submitRewards()`.